lib/compiler/src/graph_utils.h

```diff
@@ -0,0 +1,74 @@
+#ifndef _FLEXFLOW_COMPILER_GRAPH_UTILS_H
+#define _FLEXFLOW_COMPILER_GRAPH_UTILS_H
+
+#include "compiler/unity_algorithm.h"
+
+namespace FlexFlow {
+
+SerialParallelDecomposition
+    get_serial_parallel_decomposition(OptimizerPCG const &pcg);
+
+OptimizerPCG cg_to_pcg(OptimizerComputationGraph const &g);
+SubParallelComputationGraph pcg_to_subpcg(OptimizerPCG const &g);
+
+
+// NOTE(@wmdi): I think we should have the following interfaces in the graph library eventually.
```

Comment:
Sure, let's do that. Would probably be good to loosen the requirements to `Downward` and `Upward` open where it makes sense